### PR TITLE
Switch to kebab-case in site URLs

### PR DESCRIPTION
**Pull request by kachkaev**
This PR converts `camelCase` website URLs to `kebab-case` ones. This better aligns them with web standards and reduces codebase entropy in the long term. For example, `/api/uploadUserAvatar` is now `/api/upload-user-avatar`.

I have also configured [`unicorn/filename-case`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/filename-case.md) for the affected folders to make sure we keep the entropy low.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,8 +40,8 @@
     "@swc/helpers": "0.3.2",
     "@types/fs-extra": "^9.0.13",
     "@types/wait-on": "5.3.1",
-    "@typescript-eslint/eslint-plugin": "5.11.0",
-    "@typescript-eslint/parser": "5.11.0",
+    "@typescript-eslint/eslint-plugin": "5.21.0",
+    "@typescript-eslint/parser": "5.21.0",
```
Comment (kachkaev) [inline package.json:44]: Removes this warning:

```
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.6.0

YOUR TYPESCRIPT VERSION: 4.6.3

Please only submit bug reports when using the officially supported version.

=============
```

File: .eslintrc.js [hunk 2]
```diff
@@ -315,6 +319,12 @@ module.exports = {
         ],
       },
     },
+    {
+      files: ["site/src/pages/**"],
+      rules: {
+        "unicorn/filename-case": "error",
+      },
+    },
```
Comment (kachkaev) [inline .eslintrc.js:327]: Makes sure we don’t re-introduce `camelCase` page paths

File: .eslintrc.js [hunk 3]
```diff
@@ -204,6 +204,14 @@ module.exports = {
...
       },
     },
+    {
+      files: [
+        "packages/{block-scripts,block-template,blockprotocol,create-block-app}/**",
+      ],
+      rules: {
+        "unicorn/filename-case": "error",
+      },
+    },
```
Comment (kachkaev) [inline .eslintrc.js:214]: Bonus: makes sure we don’t fallback to `camelCase` in these packages. We still need to convert `mock-block-dock` and `react-block-loader` (out of scope for this PR)

**Review by CiaranMn — APPROVED** [review]
Code looks good, tested and all pages / functionality working

**Pull request merged**